Revamp Cmdpal docs (#5514)
MicrosoftDocs/windows-dev-docs: hub/powertoys/command-palette/adding-commands.md

> 

Reviewer: I can't add a comment here because I don't think you touched it in this PR... but can we change it to "Now that you've created your extension, it's time to add commands to it." and "# Add commands" (Lines 14 and 16)

To me, having "some" in there feels really informal? That could just be me though.

Author: good suggestion, the "some" isn't needed

> We can start by navigating to the `<ExtensionName>Page.cs` file. This file is the [ListPage](./microsoft-commandpalette-extensions-toolkit/listpage.md) that will be displayed when the user selects your extension. In there you should see:

Reviewer: ```suggestion
We can start by navigating to `/Pages/<ExtensionName>Page.cs`. This file is the [ListPage](./microsoft-commandpalette-extensions-toolkit/listpage.md) that will be displayed when the user selects your extension. In there you should see:
```

}
```

> Here you can see that we've set the icon for the page, the title, and the name that's shown at the top-level when you have the command selected. The `GetItems` method is where you'll return the list of commands that you want to show on this page. Right now, that's just returning a single command that does nothing. Let's instead try making that command open _this_ page in the user's default web browser.

Reviewer: ```suggestion
Here you can see that we've set the icon for the page, the title, and the name that's shown at the top-level when you have the command selected. The `GetItems` method is where you'll return the list of commands that you want to show on this page. Right now, it's just returning a single command that does nothing. Let's instead try making that command open _this_ page in the user's default web browser.
```

}
```

To update the extension in the Command Palette you need to:

1. Deploy your extension
1. In the Command Palette, type the "reload" command to refresh the extensions in the palette

> ![Screenshot of reload](../../images/command-palette/reload.png)

Reviewer 2: Alt text should be unique. This text is used three times in this file. I'll add suggestions, but feel free to use something different that makes sense to you.
```suggestion
![Screenshot of Command Palette after reloading](../../images/command-palette/reload.png)
```

Author: your right, I'll update them to be descriptive

> [!NOTE]
> There are several reload options, make sure to select the **Reload Command Palette extensions**

1. Scroll down to your extension and press `Enter`
1. Press `enter` on `Open the Command Palette documentation`

> 1. You should see that the command opens the Command Palette documentation

Reviewer: Didn't work for me :(

The **OpenUrlCommand** is a helper for opening a URL in the user's default web browser.

## Debugging Extension

As your building your extension, you'll most likely want to debug it.

1. Add a debug message to the `GetItems` function.

```diff
    public override IListItem[] GetItems()
    {
        var command = new OpenUrlCommand("https://learn.microsoft.com/windows/powertoys/command-palette/adding-commands");
    
+       Debug.Write("Debug message from GetItems");
    
        return [
            new ListItem(command)
            {
                Title = "Open the Command Palette documentation",
            }
        ];
    }
```

1. Deploy your extension, wait until it's successful

> 1. Confirm You're in Debug Configuration

Reviewer: ```suggestion
1. Confirm you're in Debug configuration
```

<details>
  <summary>Instructions to confirm debug configuration</summary>

1. Look at the toolbar at the top of Visual Studio
1. You’ll see a dropdown that says either `Debug` or `Release` (next to the green "Start" button ▶️)
1. If it says `Release`, click the dropdown and select `Debug`.

> ![Screenshot of reload](../../images/command-palette/debug-configuration.png)

Reviewer 2: ```suggestion
![Screenshot of debug configuration](../../images/command-palette/debug-configuration.png)
```

</details>

1. Run the app in debug by pressing the green "Start" button ▶️ or press `F5`
1. Ensure the Output window is set to show `Debug` output (Ctrl + Alt + O)

> ![Screenshot of reload](../../images/command-palette/output.png)

Reviewer 2: ```suggestion
![Screenshot of debug information in Output window](../../images/command-palette/output.png)
```

1. In the Command Palette, run `reload`
1. Go to your extension and select `Open the Command Palette documentation`.
1. In Visual Studio's Output window, you should see `Debug message from GetItems`

## InvokableCommand Command

Let's continue building a new command, that shows a **MessageBox**. To do that, we need to create a new class that implements `InvokableCommand`.

> 1. In Visual Studio, Add a New Class File

Reviewer: Important to reinforce where to add this file. I assumed inside of the Pages folder, but not everyone will.

Author: ack

}
```

> 1. Deploy your extension

Reviewer: Sometimes you say "Deploy your extension, wait for it to finish" or something along those lines and others you don't. Might be good to pick one for consistency. I think we can trust the user to wait for the Deploy process to finish before trying to move on.

Author: ack

> 1. In command palette, `Reload`

Reviewer: also noticing inconsistencies with how we capitalize "Command Palette"

Author: ack

And presto - a command to show a message box!

> [!TIP]
> At about this point, you'll probably want to initialize a git repo / {other source control method of your choice} for your project. This will make it easier to track changes, and to share your extension with others.
> 

> > We recommend using GitHub, as it's easy to collaborate on your extension with others, and get feedback, and share it with the world.

Reviewer: ```suggestion
> We recommend using GitHub, as it's easy to collaborate on your extension with others, get feedback, and share it with the world.
```

## Types of Pages

> So far, we've only worked with commands that "do something". However, you can also add commands that show additional pages within the Command Palette. There are basically two types of "Commands" in the Palette:

Reviewer: Is "basically" necessary?

```suggestion
So far, we've only worked with commands that "do something". However, you can also add commands that show additional pages within the Command Palette. There are two types of "Commands" in the Palette:
```
